Tests/NIOPosixTests/IPv4Header.swift

```diff
@@ -0,0 +1,365 @@
+        header.totalLength = header.totalLength.convertEndianness(to: .big)
+        // TODO: fragmentOffset needs to be in host byte order as well but it is always zero in our tests
+        // and fragmentOffset is 13 bits in size so we can't just use writeInteger(endianness: .host)
+        return self._writeIPv4Header(header)
+    }
+    
+    @discardableResult
+    mutating func writeIPv4HeaderToOSRawSocket(_ header: IPv4Header) -> Int {
+        #if canImport(Darwin)
+        self.writeIPv4HeaderToBSDRawSocket(header)
+        #else
+        self.writeIPv4Header(header)
+        #endif
+    }
+}
+
+extension IPv4Header {
+    func computeChecksum() -> UInt16 {
+        let checksum = ~[
+            UInt16(versionAndIhl) << 8 | UInt16(dscpAndEcn),
+            totalLength,
+            identification,
+            flagsAndFragmentOffset,
+            UInt16(timeToLive) << 8 | UInt16(`protocol`.rawValue),
+            UInt16(sourceIpAddress.rawValue >> 16),
+            UInt16(sourceIpAddress.rawValue & 0b0000_0000_0000_0000_1111_1111_1111_1111),
+            UInt16(destinationIpAddress.rawValue >> 16),
+            UInt16(destinationIpAddress.rawValue & 0b0000_0000_0000_0000_1111_1111_1111_1111),
+        ].reduce(UInt16(0), onesComplementAdd)
+        assert(isValidChecksum(checksum))
+        return checksum
+    }
+    mutating func setChecksum() {
+        self.headerChecksum = computeChecksum()
+    }
+    func isValidChecksum(_ headerChecksum: UInt16) -> Bool {
+        let sum = ~[
+            UInt16(versionAndIhl) << 8 | UInt16(dscpAndEcn),
+            totalLength,
+            identification,
+            flagsAndFragmentOffset,
+            UInt16(timeToLive) << 8 | UInt16(`protocol`.rawValue),
+            headerChecksum,
+            UInt16(sourceIpAddress.rawValue >> 16),
+            UInt16(sourceIpAddress.rawValue & 0b0000_0000_0000_0000_1111_1111_1111_1111),
+            UInt16(destinationIpAddress.rawValue >> 16),
+            UInt16(destinationIpAddress.rawValue & 0b0000_0000_0000_0000_1111_1111_1111_1111),
+        ].reduce(UInt16(0), onesComplementAdd)
+        return sum == 0
+    }
+    func isValidChecksum() -> Bool {
+        isValidChecksum(headerChecksum)
+    }
+}
+
+private func onesComplementAdd<Integer: FixedWidthInteger>(lhs: Integer, rhs: Integer) -> Integer {
+    var (sum, overflowed) = lhs.addingReportingOverflow(rhs)
+    if overflowed {
+        sum &+= 1
+    }
+    return sum
+}
+
+extension IPv4Header {
+    var platformIndependentTotalLengthForReceivedPacketFromRawSocket: UInt16 {
+        #if canImport(Darwin)
+        // On BSD the IP header will only contain the size of the ip packet body, not the header.
+        // This is known bug which can't be fixed without breaking old apps which already workaround the issue
+        // like e.g. we do now too.
+        return totalLength + 20
+        #elseif os(Linux)
+        return totalLength
```

Comment:
This pull [broke building the tests for Android](https://github.com/buttaface/swift-android-sdk/actions/runs/3600314592/jobs/6064844908#step:13:377). I [just confirmed that adding Android here fixes it again](https://github.com/buttaface/swift-android-sdk/pull/83/checks), buttaface/swift-android-sdk#83.

@dnadoba, mind adding these Android checks in a subsequent commit?

Reply:
Oh yeah sure, sorry about that. I think we can also just remove os(Linux) and just use an `#else`